### PR TITLE
Implemented generic kv-pair setting of export_config from args

### PR DESCRIPTION
**Pull request by borisfom**
# What does this PR do ?

This will be a basis for several export extensions that involve user input. 
As an example, I have changed handling  of --cache_support arg in export.py to use the new method.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/asr/models/asr_model.py [hunk 1]
```diff
@@ -239,3 +239,11 @@ def disabled_deployment_input_names(self):
     @property
     def disabled_deployment_output_names(self):
         return self.encoder.disabled_deployment_output_names
+
```
Comment (titu1994) [inline nemo/collections/asr/models/asr_model.py:242]: Avoid using **kwargs in user facing code, accept argument as input and pass it to super wrapped in a dict.
Plus needs docstrings

File: nemo/core/classes/exportable.py [hunk 2]
```diff
@@ -302,3 +302,11 @@ def list_export_subnets(self):
         First goes the one receiving input (input_example)
         """
         return ['self']
+
+    def get_export_config(self):
```
Comment (titu1994) [inline nemo/core/classes/exportable.py:306]: Needs docstrings

**Review by titu1994 — CHANGES_REQUESTED** [review]
Looks nice, minor changes required

File: nemo/collections/asr/models/asr_model.py (hunk 1 above)
Comment (borisfom) [reply nemo/collections/asr/models/asr_model.py:242]: This is not possible for baseclass method by design; I suppose we can do it for overloads.

File: nemo/core/classes/exportable.py (hunk 2 above)
Comment (borisfom) [reply nemo/core/classes/exportable.py:306]: Fixed.

File: nemo/collections/asr/models/asr_model.py (hunk 1 above)
Comment (borisfom) [reply nemo/collections/asr/models/asr_model.py:242]: Changed to dict argument

**Review by VahidooX — CHANGES_REQUESTED** [review]
LGTM! Would you please just add some documentation on how to export hybrid and cache-aware models to here:
https://docs.nvidia.com/deeplearning/nemo/user-guide/docs/en/main/core/export.html

File: nemo/collections/asr/models/asr_model.py (hunk 1 above)
Comment (titu1994) [reply nemo/collections/asr/models/asr_model.py:242]: Ok that is fine since we will always subclass the Exportable mixin

**Review by titu1994 — APPROVED** [review]
Looks great now, thanks !

**Comment by borisfom** [comment]
Superseded by #6983

**Pull request closed without merging**